src/overlays/actors/ovl_Obj_Lift/z_obj_lift.c

```diff
@@ -30,29 +32,182 @@ const ActorInit Obj_Lift_InitVars = {
+}
+
+void func_80B96560(ObjLift* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 sp32;
+
+    if (func_8004356C(&this->dyna) != 0) {
+        if (this->unk_16E <= 0) {
+            if (((this->dyna.actor.params >> 8) & 7) == 7) {
+                func_80B967C0(this);
+            } else {
+                sp32 = Quake_Add(ACTIVE_CAM, 1);
+                Quake_SetSpeed(sp32, 10000);
+                Quake_SetQuakeValues(sp32, 2, 0, 0, 0);
+                Quake_SetCountdown(sp32, 20);
+                func_80B9664C(this);
+            }
+            return;
+        }
+    } else {
+        this->unk_16E = D_80B969C0[((this->dyna.actor.params >> 8) & 7)];
+    }
+}
+
+void func_80B9664C(ObjLift* this) {
+    this->unk_16E = 20;
+    func_80B960D0(this, func_80B96678);
+}
+
+void func_80B96678(ObjLift* this, GlobalContext* globalCtx) {
+    if ((s32)this->unk_16E <= 0) {
+        func_80B967C0(this);
+    } else {
+        this->unk168.x = this->unk168.x + 10000;
+        this->dyna.actor.world.rot.x = (s16)(Math_SinS(this->unk168.x) * 300.0f) + this->dyna.actor.home.rot.x;
+        this->dyna.actor.world.rot.z = (s16)(Math_CosS(this->unk168.x) * 300.0f) + this->dyna.actor.home.rot.z;
+        this->dyna.actor.shape.rot.x = this->dyna.actor.world.rot.x;
+        this->dyna.actor.shape.rot.z = this->dyna.actor.world.rot.z;
+        this->unk168.y = this->unk168.y + 18000;
+        this->dyna.actor.world.pos.y = Math_SinS(this->unk168.y) + this->dyna.actor.home.pos.y;
+        this->unk168.z = this->unk168.z + 18000;
+        this->dyna.actor.world.pos.x = (Math_SinS(this->unk168.z) * 3.0f) + this->dyna.actor.home.pos.x;
+        this->dyna.actor.world.pos.z = (Math_CosS(this->unk168.z) * 3.0f) + this->dyna.actor.home.pos.z;
+    }
+    if ((this->unk_16E & 3) == 3) {
+        Audio_PlaySoundAtPosition(globalCtx, (Vec3f*)&this->dyna.actor.world, 0x10, (u16)0x2838U);
+    }
+}
+
+void func_80B967C0(ObjLift* this) {
+    func_80B960D0(&this->dyna.actor, func_80B96840);
+    Math_Vec3f_Copy(&this->dyna.actor.world.pos, &this->dyna.actor.home.pos);
+    this->dyna.actor.shape.rot = this->dyna.actor.world.rot = this->dyna.actor.home.rot;
+}
+
+void func_80B96840(ObjLift* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 bgId;
+    Vec3f sp2C;
+
+    Actor_MoveForward((Actor*)this);
+    Math_Vec3f_Copy(&sp2C, &this->dyna.actor.prevPos);
+    sp2C.y += D_80B96A10[(this->dyna.actor.params >> 1) & 1];
+    this->dyna.actor.floorHeight =
+        BgCheck_EntityRaycastFloor4(&globalCtx->colCtx, &this->dyna.actor.floorPoly, &bgId, (Actor*)this, &sp2C);
+
+    if ((D_80B96A10[(this->dyna.actor.params >> 1) & 1] - 0.001f) <=
+        (this->dyna.actor.floorHeight - this->dyna.actor.world.pos.y)) {
+        func_80B96160(this, globalCtx);
+        Audio_PlaySoundAtPosition(globalCtx, &this->dyna.actor.world.pos, 20, 0x2839U);
+        Flags_SetSwitch(globalCtx, (this->dyna.actor.params >> 2) & 0x3F);
+        Actor_Kill((Actor*)this);
```

Comment:
```suggestion
        Actor_Kill(&this->actor);
```
I may have missed some more elsewhere.